src/main/java/com/image/AllImages.java

```diff
@@ -0,0 +1,67 @@
+package com.image;
+
+import com.files.FileChecker;
+import com.log.Logs;
+
+import javax.imageio.ImageIO;
+import java.awt.image.BufferedImage;
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Logger;
+
+public class AllImages implements Images {
+
+    private List<Image> imagesList = new ArrayList<>();
+
+    @Override
+    public void addImage(Image image) {
+
+        try {
+            write(image);
+            this.imagesList.add(image);
+        } catch (IOException e) {
+            e.printStackTrace();
+
+        }
+    }
+
+    public Iterator<Image> iterator() {
+        return imagesList.iterator();
+    }
+
+
+    public void write(Image image) throws IOException {
+        Logger log = Logger.getLogger(Logs.class.getName());
+
+        if (image.getURL() != null) {
+
+            log.info("Add new url in progress for save: " + image.getURL());
+
+            Extension extension = getExtention(image.getURL());
+
+            BufferedImage bufferedImage = ImageIO.read(new URL(image.getURL()));
+
+            if (bufferedImage != null) {
+                String newFileName = Extension.path() + "\\" + image.getUser().getName() + "\\image " + System.currentTimeMillis() + "." + extension;
```

Comment:
@PavelNejinskiy please use `Paths.get` to construct path instance

Reply:
@PavelNejinskiy `Extension.path` - can we pass the base path where to save images as argument to a constructor, it is not clear why `Extension` class holds this path? Please move this constant out to current class